packages/components/src/components/hds/badge/index.ts

```diff
@@ -4,39 +4,44 @@
  */
 
 import Component from '@glimmer/component';
-import { assert } from '@ember/debug';
-
-export const DEFAULT_SIZE = 'medium';
-export const DEFAULT_TYPE = 'filled';
-export const DEFAULT_COLOR = 'neutral';
-export const SIZES = ['small', 'medium', 'large'];
-export const TYPES = ['filled', 'inverted', 'outlined'];
-export const COLORS = [
-  'neutral',
-  'neutral-dark-mode',
-  'highlight',
-  'success',
-  'warning',
-  'critical',
-];
-
-export default class HdsBadgeIndexComponent extends Component {
+import {assert} from '@ember/debug';
+import {type Color, ColorValues, type Size, SizeValues, type Type, TypeValues} from "./types.ts";
+
+export interface HdsBadgeSignature {
+  Args: {
+    size: Size;
+    type: Type;
+    color: Color;
+    text: string;
+    icon: string | null;
+    isIconOnly: boolean | null;
+  },
```

Comment:
`@text` is the only required argument, the other ones are optional. 
`@isIconOnly` is false by default, so it won't be null at any point

```suggestion
  Args: {
    size?: Size;
    type?: Type;
    color?: Color;
    text: string;
    icon?: string | null;
    isIconOnly?: boolean;
  },
```